bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/ThingProperty.java

```diff
@@ -0,0 +1,28 @@
+package org.eclipse.smarthome.core.thing;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+
+/**
+ * The {@link ThingProperty} enumeration defines all base configuration properties for a {@link Thing}. They
+ * will be injected into the {@link Configuration} of a thing during its instantiation (compare
+ * {@link BaseThingHandlerFactory#createThing(ThingTypeUID, Configuration, ThingUID, ThingUID)}). For this purpose
+ * the operation {@link BaseThingHandlerFactory#getThingProperties(Thing)} must be overwritten in the concrete thing
+ * handler factory class.
+ * 
+ * @author Thomas Höfer - initial contribution
+ */
+public enum ThingProperty {
```

Comment:
No, because this method is only in the factory, which can only deal with static values that should always be set. The enum also contains keys for non-static properties, such as the firmware version.